### ISSUE TITLE
Behat fails silently when Selenium server not running

### ISSUE THREAD

**Issue by ThomWilhelm** [Behat/Behat#885]
Apologies if this isn't something you can control.

Using behat 3.0.15 and Selenium I notice that when I run for example.

behat --format junit --profile=prod --config=behat.yml --out /tmp/build/logs

If I then shut down the selenium server and run the command again, this command just returns nothing, and all tests are skipped in the output file.

Is it possible behat can detect the server isn't running and return a helpful error message? I can crudely detect this myself by checking that if all tests have skipped, then try and contact the selenium server to ensure it's running, but if a clear error could be thrown by behat I feel that would be an improvement.

**Comment by janipalsamaki** [comment]
You might want to create a simple (shell) script that first checks that Selenium is actually running and only after that runs the tests. Something like this (adapt to your OS, shell, Selenium IP/port etc.):

```
#!/bin/bash

nc -z -v -w5 127.0.0.1 4444 > /dev/null 2>&1
if [ $? -ne 0 ]
then
  echo "Selenium not running!"
else
  echo "Running scenarios..."
  behat --format junit --profile=prod --config=behat.yml --out /tmp/build/logs
  echo "Done!"
fi
```

**Comment by ThomWilhelm** [comment]
Thanks @janipalsamaki since I'd opened this I'd already done something similar.

nmap -oG - 127.0.0.1 -p 4444  | grep 4444/open | wc -l

Am happy just to use a check such as this before calling behat.
